xarray/core/combine.py

```diff
@@ -556,29 +579,83 @@ def combine_by_coords(
     they are concatenated based on the values in their dimension coordinates,
     not on their position in the list passed to `combine_by_coords`.
 
+    >>> import pandas as pd
+    >>> import numpy as np
+    >>> import xarray as xr
+
+    >>> x1 = xr.Dataset({"temperature": (("time", "x"), 20 * np.random.rand(6).reshape(2, 3)),
+    ...                  "precipitation": (("time", "x"), np.random.rand(6).reshape(2, 3))},
+    ...                 coords={"time": pd.date_range(start="2000-01", periods=2, freq='M'),
+    ...                         "x": [10, 20, 30]})
+    >>> x2 = xr.Dataset({"temperature": (("time", "x"), 20 * np.random.rand(6).reshape(2, 3)),
+    ...                  "precipitation": (("time", "x"), np.random.rand(6).reshape(2, 3))},
+    ...                 coords={"time": pd.date_range(start="2000-03", periods=2, freq='M'),
+    ...                         "x": [10, 20, 30]})
+
+    >>> x3 = xr.Dataset({"temperature": (("time", "x"), 20 * np.random.rand(6).reshape(2, 3)),
+    ...                  "precipitation": (("time", "x"), np.random.rand(6).reshape(2, 3))},
+    ...                 coords={"time": pd.date_range(start="2000-03", periods=2, freq='M'),
+    ...                         "x": [40, 50, 60]})
```

Comment:
I sometimes find these a bit hard to read—how about formatting them with `black`? Not a top priority though

Reply:
I fixed it

Reply:
The pandas time bits aren't really necessary. Can we remove those please?

Reply:
> Can we remove those please?

Definitely